database_cleanup/models/purge_columns.py

```diff
@@ -36,9 +36,12 @@ def purge(self):
             self.env.cr.execute(
                 'SELECT count(attname) FROM pg_attribute '
                 'WHERE attrelid = '
-                '( SELECT oid FROM pg_class WHERE relname = %s ) '
-                'AND attname = %s',
-                (model_pool._table, line.name))
+                '(SELECT oid FROM pg_class WHERE relname = %(table_name)s) '
+                'AND attname = %(column_name)s AND (SELECT count(inhrelid) '
+                'FROM pg_inherits WHERE inhrelid = '
+                '(SELECT oid FROM pg_class WHERE relname = %(table_name)s )) = 0',
+                {'table_name': model_pool._table, 'column_name': line.name})
```

Comment:
Nice improvement, but maybe we can modernize these queries a little bit. I think the following is equivalent, but please check.
```
self.env.cr.execute(
    """
    select count(pa.attname)
    from pg_attribute pa
    join pg_class pc on pa.attrelid = pc.oid
    where pc.relname = %(table_name)s and pa.attname = %(column_name) 
    and not exists(select from pg_inherits where inhrelid = pc.oid);
    """,
    {'table_name': model_pool._table, 'column_name': line.name}),
```

Same for the query below, replace the subqueries.